ifrs17/Import/Importers.ipynb

```diff
@@ -1334,18 +1368,41 @@
                 "\n    if(Activity.HasErrors()) return Activity.Finish();",
                 "\n    ",
                 "\n    var hasAccidentYearColumn = dataSet.Tables[ImportFormats.Cashflow].Columns.Any(x => x.ColumnName == nameof(RawVariable.AccidentYear));",
+                "\n    var hasCashFlowPeriodicityColumn = dataSet.Tables[ImportFormats.Cashflow].Columns.Any(x => x.ColumnName == nameof(CashFlowPeriodicity));",
+                "\n    var hasInterpolationMethodColumn = dataSet.Tables[ImportFormats.Cashflow].Columns.Any(x => x.ColumnName == nameof(InterpolationMethod));",
                 "\n    ",
                 "\n    var importLog = await Import.FromDataSet(dataSet)",
                 "\n        .WithType<RawVariable> ( (dataset, datarow) => {",
                 "\n            var aocType = datarow.Field<string>(nameof(RawVariable.AocType));",
                 "\n            var novelty = datarow.Field<string>(nameof(RawVariable.Novelty));",
                 "\n            var dataNode = datarow.Field<string>(nameof(DataNode));",
-                "\n            ",
+                "\n",
+                "\n            // CashflowPeriodicity given by the Cashflows or else taken from the SingleDataNodeParameters",
+                "\n            CashFlowPeriodicity cashFlowPeriodicity = default;",
+                "\n            if (hasCashFlowPeriodicityColumn)",
+                "\n            {",
+                "\n                var cashFlowPeriodicityInput = datarow.Field<string>(nameof(SingleDataNodeParameter.CashFlowPeriodicity)); ",
+                "\n                if (Enum.TryParse(cashFlowPeriodicityInput, out CashFlowPeriodicity cfp))",
+                "\n                    cashFlowPeriodicity = cfp;",
+                "\n                //else if (!string.IsNullOrEmpty(cashFlowPeriodicityInput)) {ApplicationMessage.Log(Error.InvalidCashFlowPeriodicity, dataNode); return null;}",
+                "\n            }",
+                "\n            else parsingStorage.GetCashFlowPeriodicity(dataNode);",
```

Comment:
@dcolleoni  as requested, I checked the Parser for InterpolationMethod and Cashflow ... 
- the logic is nearly identical, 
- I aligned some naming 
- the exception is that there is a last "else Get from parsingStorage" here.
- however, I prefer not to create a Validation method in the storage, becase It would need to pass on the parsin datarow and it becomes messy... if you agree I leave the repetitive code.